src/solver/hydro/management/daily.cpp

```diff
@@ -378,6 +381,22 @@ inline void HydroManagement::prepareDailyOptimalGenerations(Solver::Variable::St
             for (uint day = firstDay; day != endDay; ++day)
             {
                 problem.TurbineMax[dayMonth] = maxP[day] * maxE[day];
+                double dailyMingen = 0.0;
```

Comment:
This a check on the daily minimum generation.
This check should not be done in a function that builds and runs a optimization problem, because it makes the function grow and be less readable (it's already too big and makes already too many things). 
This check does not need to done in the context of this function, it should be moved somewhere else (see  [this comment](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1203#discussion_r1156980396) about that, it gives a possible place to move the check).

Furthermore, we should use **std::accumulate** to compute **dailyMingen**.

I can see that the computation of **dailyMingen** is done twice in this source file, this is code duplication. This computation should done once and its result stored in **management.cpp** (follow the link above).

You could reply that there are many code duplications in this function, and you would be right.
No need to add some more.